README-3PP-LICENSE-CHECK-AND-AUDIT.md

```diff
@@ -0,0 +1,121 @@
+# VS Code builtin extensions: 3PP license check and security vulnerability audit
+
+Note: Further automation is possible for the future, but for now we will document how to do most of this semi-manually (tools are used but could be better integrated in this repo). For example we have a better integration for `dash-licenses` in the main Theia repo, and intend to make it re-usable. When that happens, we can use it in this repo, too.
+
+## "internal" builtin vscode extensions
+
+These are normally just called `vscode builtin extensions`. These are in the vscode git repository, folder `extensions`, built and packaged along `vscode`. They are not made available individually by Microsoft, so to use them with Theia, we need to this ourselves. This is what this repository is for. We have the `vscode` repository as a git submodule. We hook-into its build scripts to build the extensions and then package them individually as `.vsix` files, and publish them to [open-vsx.org](https://open-vsx.org/).
+
+### 3PP License check (builtins)
+
+We use [dash-licenses](https://github.com/eclipse/dash-licenses) to scan for 3rd party dependencies that have unclear or forbidden licenses (as per the Eclipse Foundation rules). It needs to run on each extension.
+
+```bash
+# dash-licenses - local run
+
+yarn download:dash-licenses
+export DASH_LICENSES_JAR=${PWD}/dash-licenses.jar
+# supply your own token. See: https://github.com/eclipse/dash-licenses#automatic-ip-team-review-requests
+export DASH_LICENSES_TOKEN=<token>
+
+# Run dash-licenses on each repo - a summary.txt file will be created, 
+# that contains the results
+find vscode/extensions -maxdepth 1 -mindepth 1 -type d -exec bash -c "cd '{}' &&  pwd && java -jar $DASH_LICENSES_JAR yarn.lock -timeout 120 -batch 20 -summary ./summary.txt" \;
+
```

Comment:
Good question: yes it is. The EF license scan tool, `scancode` does that. For example, see the report attached to the issue I raised on Gitlab for the "internal" builtins IP check:
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4120#note_1031903

I have submitted, for the same analysis, the external builtins that we currently use, individually (since they each have their own repo, they are easy to process this way automatically):
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4121
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4122
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4123
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4124

Reply:
> I assume you are updating the following issue accordingly?: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/4120#note_1031903

No - I added the info of what was found initially to show we would also cover the 3PP dependencies by `dash-licenses` on those extensions, but there is no real purpose in keeping an updated list. We do not keep such a list for the main repo either. 

If needed we have the means to perform a "reverse mapping" of sorts by re-running `dash-licenses` on this version of the vscode-builtins and confirming that no dependency is flagged as suspicious by `dash-license` (except `works-with` approved for us, but not yet correctly handled by the tool, like `jschardet`)